master/build-rust-manifest.py

```diff
@@ -402,8 +412,14 @@ def live_package_url(name, dist_dir, date, version, target):
     if name == "cargo":
         maybe_channel = "nightly"
 
-    url1 = s3_addy + "/" + dist_dir + "/" + date + "/" + name + "-" + version + "-" + target + ".tar.gz"
-    url2 = s3_addy + "/" + dist_dir + "/" + date + "/" + name + "-" + maybe_channel + "-" + target + ".tar.gz"
+    if name == "rust-src":
+        # The build system treats source packages as a separate target for `rustc`
+        # but for rustup we'd like to treat them as a completely separate package.
+        url1 = s3_addy + "/" + dist_dir + "/" + date + "/rustc-" + version + "-src.tar.gz"
+        url2 = s3_addy + "/" + dist_dir + "/" + date + "/rustc-" + maybe_channel + "-src.tar.gz"
```

Comment:
@Diggsey It looks to me like the only thing that needs to change is the naming scheme here. It's `rust-src-$version-or-channel.tar.gz` now.